io/include/detray/io/utils/create_path.hpp

```diff
@@ -62,15 +63,16 @@ inline std::string alt_file_name(const std::string& name) {
         }
     }
 
-    return stem + extension;
+    path = parent_path / std::filesystem::path{stem + extension};
+    return path.string();
 }
 
 /// Check if a given file path exists and generate it if not
 inline auto create_path(const std::string& outdir) {
 
     auto path = std::filesystem::path(outdir);
 
-    if (not std::filesystem::exists(path)) {
+    if (!std::filesystem::exists(path)) {
         std::error_code err;
         if (!std::filesystem::create_directories(path, err)) {
```

Comment:
```suggestion
        if (std::error_code err; !std::filesystem::create_directories(path, err)) {
```
not relevant for this pr but I thought that might be nice